### ISSUE TITLE
LaTeX in labels

### ISSUE THREAD

**Issue by basnijholt** [ioam/holoviews#67]
How can I do the following?

```python
%%opts Image [fig_latex=True]
hv.Image(np.random.randn(50,50)).relabel("$\mu$")
```

Error:
```

Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/holoviews/ipython/display_hooks.py", line 203, in wrapped
    **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/holoviews/ipython/display_hooks.py", line 229, in element_display
    info = process_object(element)
  File "/usr/local/lib/python2.7/dist-packages/holoviews/ipython/display_hooks.py", line 106, in process_object
    invalid_options = OptsMagic.process_element(obj)
  File "/usr/local/lib/python2.7/dist-packages/holoviews/ipython/magics.py", line 454, in process_element
    StoreOptions.propagate_ids(obj, tree_id, cls.applied_keys)
  File "/usr/local/lib/python2.7/dist-packages/holoviews/core/options.py", line 884, in propagate_ids
    obj.traverse(lambda o: setattr(o, 'id', new_id), specs=set(applied_keys))
  File "/usr/local/lib/python2.7/dist-packages/holoviews/core/dimension.py", line 290, in traverse
    matches = self.matches(spec)
  File "/usr/local/lib/python2.7/dist-packages/holoviews/core/dimension.py", line 259, in matches
    for ident in specification)
  File "/usr/local/lib/python2.7/dist-packages/holoviews/core/dimension.py", line 259, in <genexpr>
    for ident in specification)
  File "/usr/local/lib/python2.7/dist-packages/holoviews/core/util.py", line 169, in __call__
    name = self.remove_diacritics(name)
  File "/usr/local/lib/python2.7/dist-packages/holoviews/core/util.py", line 135, in remove_diacritics
    replacement = replacement.decode('unicode_escape')
UnicodeDecodeError: 'unicodeescape' codec can't decode byte 0x5c in position 0: \ at end of string
```

**Comment by philippjfr** [comment]
This should have been caught by some of our tests but I just pushed a fix in accc21ed466f6c2265825eabb3f98024097d4bf5. Everything else still seems to work so I don't think it will cause any issues but it would be good if @jlstevens could review it anyway.

**Comment by basnijholt** [comment]
Works, thanks!

**Comment by basnijholt** [comment]
hv.HoloMap(bands, key_dimensions=[hv.Dimension('$B_y$')]).select(y=(-10,10))

**Comment by basnijholt** [comment]
```python
hv.HoloMap(bands, key_dimensions=[hv.Dimension('$B_y$')]).select(y=(-10,10))
```
gives a lower case in the title:
![download](https://cloud.githubusercontent.com/assets/6897215/7616206/135043f8-f9a3-11e4-9720-8df4e99220a6.png)

**Comment by philippjfr** [comment]
Fixed in e94c32a069c43e0ed9bec67f0f15dbbb666ebe65, not sure how calling capitalize would cause it to become lowercase but it should be fixed now.

**Comment by jlstevens** [comment]
Note that although this is now correct for titles and dimension formatting in general, the attribute name in a ``Layout`` or ``Overlay`` will still be capitalized (in order to avoid confusion with ``Layout`` or ``Overlay`` method names).
